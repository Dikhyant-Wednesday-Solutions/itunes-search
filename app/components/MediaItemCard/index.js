import React from 'react';
import { string } from 'prop-types';
import styled from '@emotion/styled';
import { colors } from '@app/themes/index';

const Card = styled.div`
  border-style: solid;
  border-width: 1px;
  padding: 5px;
  border-radius: 5px;

  &:hover {
    cursor: pointer;
    background: ${colors.greenWhite};
  }

  &:active {
    background: ${colors.pastelGrey};
  }
`;

/**
 * MediaItemCard component that displays information about a Media item (like a song).
 *
 * @param {Object} props - The component props.
 * @param {string} props.trackName - The name of the track.
 * @param {string} props.collectionName - The name of the collection.
 * @param {string} props.artistName - The name of the artist.
 * @param {string} props.country - The name of the country the media is from
 * @param {string} props.primaryGenreName - Genre name
 * @param {string} props.thumbnailSrc - thumbnail of the media
 * @returns {JSX.Element} The MediaItemCard component displaying the song information.
 */
export function MediaItemCard({ trackName, collectionName, artistName, country, primaryGenreName, thumbnailSrc }) {
  return (
    <Card data-testid="media-item-card">
      <img data-testid="thumbnail" src={thumbnailSrc} alt="thumbnail" />
      <div data-testid="track_name">{trackName}</div>
      <div data-testid="collection_name">{collectionName}</div>
      <div data-testid="artist_name">{artistName}</div>
      <div data-testid="country">{country}</div>
      <div data-testid="primary_genre_name">{primaryGenreName}</div>
    </Card>
  );
}

MediaItemCard.propTypes = {
  trackName: string,
  collectionName: string,
  artistName: string,
  country: string,
  primaryGenreName: string,
  thumbnailSrc: string
};

export default MediaItemCard;
