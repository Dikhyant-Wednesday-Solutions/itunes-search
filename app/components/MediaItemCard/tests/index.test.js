/**
 *
 * Tests for MediaItemCard
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import MediaItemCard from "../index";

describe('<MediaItemCard />', () => {
    it('should render and match the snapshot', () => {
      const { baseElement } = renderWithIntl(<MediaItemCard />);
      expect(baseElement).toMatchSnapshot();
    });

    it('should contain 1 MediaItemCard component', () => {
        const { getAllByTestId } = renderWithIntl(<MediaItemCard />);
        expect(getAllByTestId('media-item-card').length).toBe(1);
    });

    it('should render the media details inside the card', () => {
        const artistName = "J.J. Abrams";
        const trackName = "Star Wars: The Force Awakens";
        const collectionName = "Star Wars";
        const country = "USA";
        const primaryGenreName = "Action / Adventure";
        const thumbnailSrc = "https://is1-ssl.mzstatic.com/image/thumb/Video123/v4/1f/2b/ae/1f2bae7f-62a1-1055-8471-401291b6dcdd/pr_source.lsr/100x100bb.jpg";
        const { getByTestId } = renderWithIntl(
            <MediaItemCard
                artistName={artistName}
                trackName={trackName}
                collectionName={collectionName}
                country={country}
                primaryGenreName={primaryGenreName}
                thumbnailSrc={thumbnailSrc}
            />
        );

        expect(getByTestId('thumbnail')).toHaveAttribute('src', thumbnailSrc);
        expect(getByTestId('track_name')).toHaveTextContent(trackName);
        expect(getByTestId('collection_name')).toHaveTextContent(collectionName);
        expect(getByTestId('artist_name')).toHaveTextContent(artistName);
        expect(getByTestId('country')).toHaveTextContent(country);
        expect(getByTestId('primary_genre_name')).toHaveTextContent(primaryGenreName);
 });
});