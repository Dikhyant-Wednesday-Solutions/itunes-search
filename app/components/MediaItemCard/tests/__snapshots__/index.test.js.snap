// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaItemCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="css-17i9qj4 e1f5w7bs4"
      data-testid="media-item-card"
    >
      <img
        alt="thumbnail"
        data-testid="thumbnail"
        id="thumbnail"
        style="position: absolute; top: 0px; width: 100%;"
      />
      <div
        class="css-a6o9mb e1f5w7bs0"
      >
        <div
          class="css-d3v9zr e1f5w7bs3"
        >
          <div
            class="css-l8l8b8 e1f5w7bs2"
            data-testid="track_name"
            style="color: white;"
          />
          <div
            class="css-l8l8b8 e1f5w7bs2"
            data-testid="artist_name"
            style="color: white;"
          />
        </div>
        <div
          class="css-12q6kdw e1f5w7bs1"
        >
          <button
            class="css-3x49ea e11o46g22"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium e11o46g21 css-2pnv19-MuiSvgIcon-root"
              data-testid="PlayArrowOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 8.64 15.27 12 10 15.36zM8 5v14l11-7z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
